### DISCUSSION TITLE
Allow for typed block collections

### DISCUSSION THREAD

**Discussion by knownasred** [payloadcms/payload#2617] [Feature Requests & Ideas]
[Related discord discussion](https://discord.com/channels/967097582721572934/1102152070750011402)

I have the same blocks present within multiple Field across collections, and while in the GraphQL output, the Blocks are typed with an explicit notation, it is not the case for the typescript ones, that instead in-lines all the possibilities.
Is there a reason for that, or just an oversight?

---
This seems like an interesting feature to add, as my current solution is to generate typings manually for the block types, and using them on the frontend.

**Comment by rzeczkowskip** [comment]
As a temporary solution I'm using this:

```
type Blocks = NonNullable<Page['blocks']>;
type SpecificBlock = Extract<Blocks[number], { blockType: 'specific-block' }>;
```

Might be helpful for now.
